spyderlib/plugins/ipythonconsole.py

```diff
@@ -1058,8 +1058,6 @@ def _create_client_for_kernel(self, cf, hostname, kf, pw):
         # Verifying if the connection file exists
         cf = osp.basename(cf)
         try:
-            if not cf.startswith('kernel') and not cf.endswith('json'):
-                cf = to_text_string('kernel-' + cf + '.json')

```

Comment:
It has been the case for a while AFAIK. Here is the function [find_conneciton_file](https://github.com/ipython/ipython/blob/0.13.x/IPython/lib/kernel.py#L56) in 0.13.2.

*If the argument does not match an existing file, it will be interpreted as a fileglob, and the matching file in the profile's security dir with the latest access time will be used.*